sandbox/tests/storage_test.rs

```diff
@@ -73,3 +73,91 @@ fn test_ext_storage_set(){
     assert_eq!(result, [0u8; 0]);
     assert_eq!(exp_value, Some(value.encode()));
 }
+
+#[test]
+fn test_ext_storage_read(){
+    let setup = Setup::new();
+    let mut ext = setup.ext;
+    let mut ext = ext.ext();
+
+    let key = b"great".to_vec();
+    let value: [u8; 25] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2];
+    let offset: u32 = 10;
+    ext.set_storage(key.encode(), value.to_vec());
+
+    let mut arg = vec![];
+    arg.extend(key.encode());
+    arg.extend(offset.encode());
+    let result = call_in_wasm(
+        "test_storage_read", 
+        &arg,
+        WasmExecutionMethod::Interpreted,
+        &mut ext
+    ).unwrap();
+    let res: u32 = <u32>::decode(&mut result.as_ref()).unwrap();
+    println!("{:?}", arg);
+    println!("{:?}", result);
+    assert!(res >= offset);
+    assert_eq!(&result[4..result.len()], &[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 0, 0, 0, 0, 0]);
+}
+#[test]
+#[should_panic]
```

Comment:
Why this should fail? is it because of the `ext.storage($key).unwrap()`

Reply:
yes, when you unwrap() vector with None, it should panic

Reply:
What if we explicitly check for none? It won't throw then.
My concern is that it may panic because we are calling the "test_storage_exists" function in WASM  (completely different reason) and the test will pass